Tests/Marketplace/marketplace_services.py

```diff
@@ -640,18 +650,25 @@ def _load_pack_dependencies(self, index_folder_path, pack_names):
             dependency_metadata_path = os.path.join(index_folder_path, dependency_pack_id, Pack.METADATA)
 
             if os.path.exists(dependency_metadata_path):
+                # Case 1: the dependency is found in the index.zip
                 with open(dependency_metadata_path, 'r') as metadata_file:
                     dependency_metadata = json.load(metadata_file)
                     dependencies_data_result[dependency_pack_id] = dependency_metadata
+
             elif dependency_pack_id in pack_names:
-                # If the pack is dependent on a new pack (which is not yet in the index.json)
-                # we will note that it is missing dependencies.
-                # And finally after updating all the packages in index.json.
-                # We will go over the pack again to add what was missing
-                self._is_missing_dependencies = True
-                logging.warning(f"{self._pack_name} pack dependency with id {dependency_pack_id} "
-                                f"was not found in index, marking it as missing dependencies - to be resolved in next"
-                                f" iteration over packs")
+                if id_set:
+                    if marketplace not in id_set.get('Packs', {}).get(dependency_pack_id, {}).get('marketplaces'):
+                        # Case 2: the dependency is not in the index since it is not a part of the current marketplace
+                        logging.warning(f"{self._pack_name} pack dependency with id {dependency_pack_id} is not part of"
+                                        f" the current marketplace, ignoring dependency (this is probably an optional "
```

Comment:
Do we have a way here to make sure it is optional?

Reply:
removed comment as discussed